src/databricks/labs/ucx/source_code/dependencies.py

```diff
@@ -140,15 +267,17 @@ def __init__(
         self,
         dependency: Dependency,
         parent: DependencyGraph | None,
-        loader: DependencyLoader,
-        resolver: DependencyResolver | None = None,
+        resolver: DependencyResolver,
     ):
         self._dependency = dependency
         self._parent = parent
-        self._loader = loader
-        self._resolver = resolver or DependencyResolver()
+        self._resolver = resolver
         self._dependencies: dict[Dependency, DependencyGraph] = {}
 
+    @property
+    def resolver(self) -> DependencyResolver:
```

Comment:
don't expose private properties, but rather inject those as dependencies via constructor to relevant places

Reply:
can't really do that for RunCell, but I changed this